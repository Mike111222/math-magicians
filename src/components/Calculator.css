.calculator {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 40px;
}

.screen {
  width: 98%;
  height: 60px;
  line-height: 60px;
  background-color: #ddd;
  text-align: right;
  font-size: 24px;
  padding: 5px;
  display: flex;
  justify-content: flex-end;
  align-items: center;
}

.row-div {
  display: flex;
  justify-content: space-between;
  width: 280px;
  margin-bottom: 1px;
}

.button {
  width: 65px;
  height: 65px;
  font-size: 24px;
  background-color: #f1f1f1;
  border: none;
  border-radius: 5px;
  cursor: pointer;
}

.btn-clear {
  background-color: #eaeaea;
}

.btn-operator {
  background-color: #ff9800;
  color: white;
}

.button-zero {
  width: 135px;
}

.btn-equal {
  background-color: #ff9800;
  color: white;
  height: 140px;
}
