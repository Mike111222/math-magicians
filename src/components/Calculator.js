import React from 'react';
import './Calculator.css';

function Calculator() {
  return (
    <div className="calculator">
      <div className="row-div">
        <div className="screen">0</div>
      </div>
      <div className="row-div">
        <div className="col-3">
          <button type="button" className="button btn-clear">AC</button>
        </div>
        <div className="col-3">
          <button type="button" className="button btn-clear">+/-</button>
        </div>
        <div className="col-3">
          <button type="button" className="button btn-clear">%</button>
        </div>
        <div className="col-3">
          <button type="button" className="button btn-operator">รท</button>
        </div>
      </div>
      <div className="row-div">
        <div className="col-3">
          <button type="button" className="button">7</button>
        </div>
        <div className="col-3">
          <button type="button" className="button">8</button>
        </div>
        <div className="col-3">
          <button type="button" className="button">9</button>
        </div>
        <div className="col-3">
          <button type="button" className="button btn-operator">x</button>
        </div>
      </div>
      <div className="row-div">
        <div className="col-3">
          <button type="button" className="button">4</button>
        </div>
        <div className="col-3">
          <button type="button" className="button">5</button>
        </div>
        <div className="col-3">
          <button type="button" className="button">6</button>
        </div>
        <div className="col-3">
          <button type="button" className="button btn-operator">-</button>
        </div>
      </div>
      <div className="row-div">
        <div className="col-3">
          <button type="button" className="button">1</button>
        </div>
        <div className="col-3">
          <button type="button" className="button">2</button>
        </div>
        <div className="col-3">
          <button type="button" className="button">3</button>
        </div>
        <div className="col-3">
          <button type="button" className="button btn-operator">+</button>
        </div>
      </div>
      <div className="row-div">
        <div className="col-6">
          <button type="button" className="button button-zero">0</button>
        </div>
        <div className="col-3">
          <button type="button" className="button">.</button>
        </div>
        <div className="col-3">
          <button type="button" className="button btn-operator">=</button>
        </div>
      </div>
    </div>
  );
}

export default Calculator;
