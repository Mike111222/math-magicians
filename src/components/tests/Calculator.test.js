import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Calculator from '../Calculator';

describe('Calculator', () => {
  it('should multiply two numbers correctly', () => {
    const { getByText, getByTestId } = render(<Calculator />);
    fireEvent.click(getByText('2'));
    fireEvent.click(getByText('x'));
    fireEvent.click(getByText('3'));
    fireEvent.click(getByText('='));
    expect(getByTestId('screen')).toHaveTextContent('6');
  });

  // Write more test cases for other functionalities of the Calculator component

  // You can write similar tests for other React components
});
