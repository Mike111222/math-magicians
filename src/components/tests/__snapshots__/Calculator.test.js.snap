// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Page renders correctly 1`] = `
<DocumentFragment>
  <div
    class="calculator"
  >
    <div
      class="row-div"
    >
      <div
        class="screen"
      >
        0
      </div>
    </div>
    <div
      class="row-div"
    >
      <div
        class="col-3"
      >
        <button
          class="button btn-clear"
          type="button"
        >
          AC
        </button>
      </div>
      <div
        class="col-3"
      >
        <button
          class="button btn-clear"
          type="button"
        >
          +/-
        </button>
      </div>
      <div
        class="col-3"
      >
        <button
          class="button btn-clear"
          type="button"
        >
          %
        </button>
      </div>
      <div
        class="col-3"
      >
        <button
          class="button btn-operator"
          type="button"
        >
          รท
        </button>
      </div>
    </div>
    <div
      class="row-div"
    >
      <div
        class="col-3"
      >
        <button
          class="button "
          type="button"
        >
          7
        </button>
      </div>
      <div
        class="col-3"
      >
        <button
          class="button "
          type="button"
        >
          8
        </button>
      </div>
      <div
        class="col-3"
      >
        <button
          class="button "
          type="button"
        >
          9
        </button>
      </div>
      <div
        class="col-3"
      >
        <button
          class="button btn-operator"
          type="button"
        >
          x
        </button>
      </div>
    </div>
    <div
      class="row-div"
    >
      <div
        class="col-3"
      >
        <button
          class="button "
          type="button"
        >
          4
        </button>
      </div>
      <div
        class="col-3"
      >
        <button
          class="button "
          type="button"
        >
          5
        </button>
      </div>
      <div
        class="col-3"
      >
        <button
          class="button "
          type="button"
        >
          6
        </button>
      </div>
      <div
        class="col-3"
      >
        <button
          class="button btn-operator"
          type="button"
        >
          -
        </button>
      </div>
    </div>
    <div
      class="row-div"
    >
      <div
        class="col-3"
      >
        <button
          class="button "
          type="button"
        >
          1
        </button>
      </div>
      <div
        class="col-3"
      >
        <button
          class="button "
          type="button"
        >
          2
        </button>
      </div>
      <div
        class="col-3"
      >
        <button
          class="button "
          type="button"
        >
          3
        </button>
      </div>
      <div
        class="col-3"
      >
        <button
          class="button btn-operator"
          type="button"
        >
          +
        </button>
      </div>
    </div>
    <div
      class="row-div"
    >
      <div
        class="col-6"
      >
        <button
          class="button button-zero"
          type="button"
        >
          0
        </button>
      </div>
      <div
        class="col-3"
      >
        <button
          class="button "
          type="button"
        >
          .
        </button>
      </div>
      <div
        class="col-3"
      >
        <button
          class="button btn-operator"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
